### DISCUSSION TITLE
Is NTWindowsGenesis needed when using ImageMagick as a lib on Windows?

### DISCUSSION THREAD

**Discussion by Cyriuz** [ImageMagick/ImageMagick#6152] [Development] [answered]
### ImageMagick version

7.1.0-57

### Operating system, version and so on

Windows 11

### Description

I'm using ImageMagick as a library and had some issues with our crash reporting integration after integrating ImageMagick. It turns out ImageMagick uses the SetUnhandledExceptionFilter which wipes out the filter set by the crash reporting. This is fairly simple to work around, just changing the order of things, but it got me thinking a bit about the code in nt-base.c and other windows specific code. Would there be any big downside to not use nt-base at all? And in that case would it make sense to add options to disable it? I even tested undeffing MAGICKCORE_WINDOWS_SUPPORT and the code built with visual studio 2022 but it is used in many places so that might be a bit extreme,

**Comment by snibgo** [comment]
I can't answer the question.

I can say that in my builds of IM, using Cygwin tools on Window 8.1 and Windows 11, MAGICKCORE_WINDOWS_SUPPORT and WIN32 and WIN64 are not defined.

**Comment by urban-warrior** [comment, marked as answer]
We likely should add the SetUnhandledExceptionFilter in main() rather than in the API.  We will review and add a patch within a week.

**Comment by Cyriuz** [comment]
Thanks for the fix! I saw that you made a new release, when does that become available as a github release?
